### DISCUSSION TITLE
portable-pty: how to stream dynamic changes?

### DISCUSSION THREAD

**Discussion by jjangga0214** [wezterm/wezterm#3939] [Q&A] [answered]
Hi! 

I am using portable-pty. 
Some CLIs show dynamic outputs that change as time elapses.
For example, the loading icon rotates. The downloads bar grows. A label's text and color changes.
How can I just stream(realtime) them without losing those dynamic effects?

Thanks :)

**Comment by wez** [comment, marked as answer]
A PTY is a thin layer over a pipe-like kernel construct, with a couple of extra functions for managing things like window size.
It is not inherently stateful.

It sounds like you may actually want a terminal emulator rather than a PTY.

Here's a link to a thread on matrix/element with some suggestions and other context:

https://matrix.to/#/!PirwUBcuIlTXwNveYz:matrix.org/$jK3yu_RkQzV4Gvw-x8I3BeKCYxuduGnQ8aE5Og3ZNq8?via=matrix.org&via=kde.org&via=tchncs.de
